packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -492,6 +492,35 @@ export type InsertableField<TSchema extends ImplicitFieldSchema | UnsafeUnknownS
 		? InsertableContent | undefined
 		: never;
 
+/**
+ * Content which could be read from a field within a tree.
+ *
+ * @remarks
+ * Extended version of {@link TreeFieldFromImplicitField} that also allows {@link (UnsafeUnknownSchema:type)}.
+ * Since reading from fields with non-exact schema is still safe, this is only useful (compared to {@link TreeFieldFromImplicitField}) when the schema is also used as input and thus allows {@link (UnsafeUnknownSchema:type)}
+ * for use
+ * @system @alpha
+ */
+export type ReadableField<TSchema extends ImplicitFieldSchema | UnsafeUnknownSchema> =
+	TSchema extends ImplicitFieldSchema
+		? TreeFieldFromImplicitField<TSchema>
+		: TreeLeafValue | TreeNode;
+
+/**
+ * Content which could be read from a field within a tree.
+ *
+ * @remarks
+ * Extended version of {@link TreeFieldFromImplicitField} that also allows {@link (UnsafeUnknownSchema:type)}.
+ * Since reading from fields with non-exact schema is still safe, this is only useful (compared to {@link TreeFieldFromImplicitField}) when the schema is also used as input and thus allows {@link (UnsafeUnknownSchema:type)}
+ * for use
+ * @system @alpha
+ */
+export type ReadSchema<TSchema extends ImplicitFieldSchema | UnsafeUnknownSchema> = [
+	TSchema,
+] extends [ImplicitFieldSchema]
```

Comment:
That is a neat and interesting trick to get the variance right

Reply:
This case is more subtle than that. While it comes right out of the official docs for conditionals ( https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types ) we are not using it for variance here.

Omitting the pattern here isn't really for variance reasons. Its included here to enable TypeScript to prove that ReadSchema<T> is identical to T when `TSchema extends ImplicitFieldSchema` . With out the pattern here, if the schema were a union, this identity would not hold (at least TS can't prove it), and that prevents implementing the non UnsafeUnknownSchema interfaces in a generic context where TSchema extends ImplicitFieldSchema

Reply:
Very interesting, thanks. I think I get it.